### PR TITLE
Separated sitemaps by language

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: docs/tests.py [hunk 1]
```diff
@@ -239,11 +239,23 @@ def test_empty_documents(self):
...
+    @classmethod
+    def tearDownClass(cls):
+        # cleanup URLconfs changed by django-hosts
+        set_urlconf(None)
+        super().tearDownClass()
+
+    def test_sitemap_index(self):
+        response = self.client.get('/sitemap.xml', HTTP_HOST='docs.djangoproject.dev:8000')
+        self.assertContains(response, '<sitemap>', count=2)
+        self.assertContains(response, "<loc>http://docs.djangoproject.dev:8000/sitemap-en.xml</loc>")
```
Comment (timgraham) [inline docs/tests.py]: single quotes please

File: djangoproject/urls/docs.py [hunk 2]
```diff
@@ -1,15 +1,47 @@
...
+    _data = {}
+
+    def __iter__(self):
+        return iter(
+            DocumentRelease.objects.values_list('lang', flat=True).distinct().order_by('lang')
+        )
+
+    def __getitem__(self, key):
+        if key not in self._data:
+            if not DocumentRelease.objects.filter(lang=key).exists():
```
Comment (timgraham) [inline djangoproject/urls/docs.py]: Could test coverage be improved a bit? For example, no tests are failing with this block commented out (didn't do an extensive audit).

File: docs/views.py [hunk 3]
```diff
@@ -294,3 +296,14 @@ def search_description(request, lang, version):
 if not settings.DEBUG:
     # 1 week because there is no need to render it more often
     search_description = cache_page(60 * 60 * 24 * 7)(search_description)
+
+
+@x_robots_tag
+def sitemap_index(request, sitemaps):
+    """Simplified django.contrib.sitemaps.views.index to use django_hosts reversion"""
```
Comment (timgraham) [inline docs/views.py]: Not sure about "django_hosts reversion" means. Is it a typo for "version"?

File: docs/views.py [hunk 4]
```diff
@@ -294,3 +296,14 @@ def search_description(request, lang, version):
...
+    sites = []
+    for section in sitemaps.keys():
+        sitemap_url = reverse('document-sitemap', host='docs', kwargs={'section': section})
+        sites.append(sitemap_url)
+    return TemplateResponse(request, 'sitemap_index.xml', {'sitemaps': sites},
+                            content_type='application/xml')
```
Comment (timgraham) [inline docs/views.py]: use single line or hanging indent please

**Comment by claudep** [comment]
> I'm not too familiar with sitemaps. Could you explain what this change accomplishes?

We currently have a big sitemap containing all original and translated URLs mixed.
https://docs.djangoproject.com/sitemap.xml

The idea of this patch was to provide a short index sitemap and create a separate sitemap for each language (see https://www.sitemaps.org/protocol.html#index). If you think it's not needed, we can drop the patch.

**Comment by timgraham** [comment]
I don't have any objection to the patch. It sounds like an advantage is not having to download a single huge file.

File: docs/views.py (hunk 3 above)
Comment (claudep) [inline docs/views.py]: This is my neologism for `reverse()`ion!

Comment (shaib) [inline docs/views.py]: Note that the term "reversion" is already being used in the Django sphere, by a widely used package: https://django-reversion.readthedocs.io/en/stable/

Comment (timgraham) [inline docs/views.py]: Suggested wording:
```
"""
Simplified version of django.contrib.sitemaps.views.index that uses
django_hosts for URL reversing.
"""
```

**Pull request merged**